src/utils/store-utils.ts

```diff
@@ -81,3 +82,37 @@ export async function fetchConfig(
     }
   }
 }
+
+export async function readFile(file: File): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
+    const reader = new FileReader();
+
+    reader.onload = (e) => {
+      const content = e.target?.result as string;
+      resolve(content);
+    };
+
+    reader.onerror = (error) => {
+      reject(error);
+    };
+
+    reader.readAsText(file);
+  });
+}
+
+export function parseTsvContent(content: string): { headers: string[]; data: string[][] } {
+  if (!content) return { headers: [], data: [] };
+
+  // TODO: simply skipping empty rows here may cause downstream problems,
+  // see: https://github.com/neurobagel/annotation-tool/issues/142
```

Comment:
great catch, slight suggestion so we don't have to open the link first to understand
```suggestion
  // TODO: consider validating the table against our schema first,
  // simply skipping empty rows here may cause downstream problems,
  // see: https://github.com/neurobagel/annotation-tool/issues/142
```